script.program.homeassistant/addon.py

```diff
@@ -0,0 +1,463 @@
+	file = open(fileHakaFavourites, open_param)
+	file.write(yaml.dump(entity_list))
+	file.close()
+
+def importDomainSettings():
+	if __addon__.getSetting('importAutomation') == 'true':
+		haDomainSettings[0] = True
+	if __addon__.getSetting('importClimate') == 'true':
+		haDomainSettings[1] = True
+	if __addon__.getSetting('importGroups') == 'true':
+		haDomainSettings[2] = True
+	if __addon__.getSetting('importLights') == 'true':
+		haDomainSettings[3] = True
+	if __addon__.getSetting('importScenes') == 'true':
+		haDomainSettings[4] = True
+	if __addon__.getSetting('importScripts') == 'true':
+		haDomainSettings[5] = True
+	if __addon__.getSetting('importSensors') == 'true':
+		haDomainSettings[6] = True
+	if __addon__.getSetting('importSwitches') == 'true':
+		haDomainSettings[7] = True
+	if __addon__.getSetting('importVacuums') == 'true':
+		haDomainSettings[8] = True
+	if __addon__.getSetting('importPersons') == 'true':
+		haDomainSettings[9] = True
+	log('Domain settings imported: ' + str(haDomainSettings))
+
+def getRequest(api_ext):
+	try:
+		log('Trying to make a get request to ' + api_base + api_ext)
+		r = requests.get(api_base + api_ext, headers=headers)
+		log(r.json())
+		return r.json()
+	except:
+		log('Status code error is: ' + str(r.raise_for_status()))
+		show_dialog(__addon__.getLocalizedString(30052)) #Unknown error: Check IP address or if server is online	
+		log('GetRequest status code is: ' + str(r.status_code))		
+		if r.status_code == 401:
+			show_dialog(__addon__.getLocalizedString(30050)) #Error 401: Check your token
+		elif r.status_code == 405:
+			show_dialog(__addon__.getLocalizedString(30051)) #Error 405: Method not allowed
+		elif r.status_code == 200:
+			return r
+
+
+def postRequest(api_ext, entity_id):
+	try:
+		payload = "{\"entity_id\": \"" + entity_id + "\"}"
+		log('Trying to make a post request to ' + api_base + api_ext + ' with payload: ' + payload)
+		r = requests.post(api_base + api_ext, headers=headers, data=payload)
+		log('GetRequest status code is: ' + str(r.status_code))
+		if r.status_code == 401:
+			show_dialog(__addon__.getLocalizedString(30050)) #Error 401: Check your token
+		elif r.status_code == 405:
+			show_dialog(__addon__.getLocalizedString(30051)) #Error 405: Method not allowed
+	except:
+		show_dialog(__addon__.getLocalizedString(30052)) #Unknown error: Check IP address or if server is online
+		
+def browseByDomain():
+	log('Browse by domain started')
+	listing=[]
+	isFolder = True
+
+	#Test connection / get version
+	response = getRequest('/config')
+	if response is not None:
+		log('Response from server: ' + str(response))
+	else:
+		log('Response from server is NONE!')
+
+	#Compose folder list based on addon settings
+	for d in range(len(haDomainSettings)):
+		if haDomainSettings[d]:
+			url = build_url({'mode': 'loadfolder', 'type': 'domain', 'domain': haDomainNames[d]})
+			icon = os.path.join(imgIconResourcePath) + '\\' + haDomainNames[d] + '.png'
```

Comment:
have a look into all your os.path.join definitions cause they are wrong (and will not work across platforms:

```python
icon = os.path.join(imgIconResourcePath, haDomainNames[d] + '.png')
```